### PR TITLE
feat: LSDV-5104: Split Ranker into List+Ranker

### PR DESCRIPTION
**Pull request by hlomzik**
Back to our normal architecture:
- List is object tag to display group of items
- Ranker is control tag, attached to List to make it interactive
  - In simple case it allows to reorder items of List tag
  - With Buckets inside it allows to group items

### PR fulfills these requirements
- [ ] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [ ] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Frontend

### Describe the reason for change
_(link to issue, supportive screenshots etc.)_

#### What does this fix?
_(if this is a bug fix)_

#### What is the new behavior?
_(if this is a breaking or feature change)_

#### What is the current behavior?
_(if this is a breaking or feature change)_

#### What libraries were added/updated?
_(list all with version changes)_

#### Does this change affect performance?
_(if so describe the impacts positive or negative)_

#### Does this change affect security?
_(if so describe the impacts positive or negative)_

#### What alternative approaches were there?
_(briefly list any if applicable)_

#### What feature flags were used to cover this change?
_(briefly list any if applicable)_

### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit

### Which logical domain(s) does this change affect?
_(for bug fixes/features, be as precise as possible. ex. Authentication, Annotation History, Review Stream etc.)_

### CODE REVIEW

File: src/tags/object/List.js [hunk 1]
```diff
@@ -101,47 +80,27 @@ const Model = types
...
   observer(({ item }) => {
     const data = item.dataSource;
 
     if (!data) return null;
+    // Ranker tag will display all items in interactive mode
+    if (item.ranker) return null;
 
     return (
-      <Ranker inputData={data} handleChange={item.updateResult} />
+      <React.StrictMode>
```
Comment (bmartel) [inline src/tags/object/List.js:96]: Why the inclusion of StrictMode here?

Comment (hlomzik) [reply src/tags/object/List.js:92]: Honestly I don't know... There are no errors in their implementation in strict mode, so yes, we can just remove it. And this strict wrapper as well. thanks!

File: src/components/Ranker/Ranker.tsx [hunk 2]
```diff
@@ -1,126 +1,109 @@
...
-        ...data.columns,
-        [newStartCol.id]: newStartCol,
-        [newEndCol.id]: newEndCol,
-      },
+      itemIds: newItemIds,
     };
-    //update results
 
-    handleChange ? handleChange(newData.columns[`${newData.columnOrder.length - 1}`].itemIds) : null;
+    console.log('HANDLE DRAG', newData, newItemIds);
```
Comment (bmartel) [inline src/components/Ranker/Ranker.tsx:87]:
```suggestion
```

File: src/tags/control/Ranker.js [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+    },
+  }));
+
+const RankerModel = types.compose('RankerModel', Base, AnnotationMixin, Model);
+
+const HtxRanker = inject('store')(
+  observer(({ item }) => {
+    const data = item.dataSource;
+
+    console.log('DATA', data);
```
Comment (bmartel) [inline src/tags/control/Ranker.js:192]:
```suggestion
```

File: src/tags/control/Ranker.js [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+    get dataSource() {
+      const data = self.list?._value;
+      const items = self.list?.items;
+      const ids = Object.keys(items);
+      const columns = self.columns;
+      /** @type {{ [bucket: string]: string[] }} */
+      const result = self.result?.value.ranker;
+      let itemIds = {};
+
+      console.log('PRE DATA', result, self.leftInList);
```
Comment (bmartel) [inline src/tags/control/Ranker.js:119]:
```suggestion
```

File: src/components/Ranker/Ranker.tsx [hunk 5]
```diff
@@ -1,126 +1,108 @@
...
 import React, { useEffect, useState } from 'react';
-import { DragDropContext } from 'react-beautiful-dnd';
+import { DragDropContext, DropResult } from 'react-beautiful-dnd';
 
-/**
- * components
- */
 import Column from './Column';
-import { BoardData, ColumnData } from './createData';
+import { ColumnData, NewBoardData } from './createData';
```
Comment (bmartel) [inline src/components/Ranker/Ranker.tsx:5]:
```suggestion
import { NewBoardData } from './createData';
```

File: src/components/Ranker/Ranker.module.scss [hunk 6]
```diff
@@ -1,45 +1,14 @@
...
-}
-
-.columnsStyle {
+.board {
   display: flex;
   justify-content: space-evenly;
   width: 100%;
+  padding: 10px;
+  background-color: lightgray;
+  border: 1px solid rgba(34, 36, 38, 0.15);
```
[the remark this answers is not in the document]
Comment (hlomzik) [reply src/components/Ranker/Ranker.module.scss]: Thanks for reminder! I improved styles, made them more light-weight and consistent, added css vars for colors and gaps.

File: src/tags/object/List.js (hunk 1 above)
Comment (bmartel) [reply src/tags/object/List.js:91]: Are we leaving this on, or removing this?

**Review by bmartel — APPROVED** [review]

**Pull request merged**